analysis/dataflow/mark.go

```diff
@@ -76,6 +87,22 @@ func (m MarkType) String() string {
 	}
 }
 
+// MarkIndex wraps an index kind and the index value together
+type MarkIndex struct {
+	// Kind is the kind of index (currently, either a real index or not)
```

Comment:
Do we plan to have other kinds of indices?

Reply:
My thinking was that some point we might want to track indices resulting from `next` and `select` instructions as a special case.